src/transformers/models/ibert/modeling_ibert.py

```diff
@@ -0,0 +1,1418 @@
+        encoder_hidden_states=None,
+        encoder_attention_mask=None,
+        past_key_value=None,
+        output_attentions=False,
+    ):
+        assert past_key_value is None
+        self_attn_past_key_value = past_key_value[:2] if past_key_value is not None else None
+        self_attention_outputs, self_attention_outputs_scaling_factor = \
+            self.attention(
+                hidden_states,
+                hidden_states_scaling_factor,
+                attention_mask,
+                head_mask,
+                output_attentions=output_attentions,
+                past_key_value=self_attn_past_key_value,
+            )
+        attention_output = self_attention_outputs[0]
+        attention_output_scaling_factor = self_attention_outputs_scaling_factor[0]
+
+        outputs = self_attention_outputs[1:]  # add self attentions if we output attention weights
+        outputs_scaling_factor = self_attention_outputs_scaling_factor[1:]  
+
+        layer_output, layer_output_scaling_factor = \
+                self.feed_forward_chunk(attention_output, attention_output_scaling_factor)
+        outputs = (layer_output,) + outputs
+
+        return outputs
+
+    def feed_forward_chunk(self, attention_output, attention_output_scaling_factor):
+        attention_output, attention_output_scaling_factor = \
+                self.pre_intermediate_act(attention_output, attention_output_scaling_factor)
+        intermediate_output, intermediate_output_scaling_factor = \
+                self.intermediate(attention_output, attention_output_scaling_factor)
+
+        intermediate_output, intermediate_output_scaling_factor = \
+                self.pre_output_act(intermediate_output, intermediate_output_scaling_factor)
+        layer_output, layer_output_scaling_factor = self.output(
+                intermediate_output, intermediate_output_scaling_factor,
+                attention_output, attention_output_scaling_factor)
+        return layer_output, layer_output_scaling_factor
+
+
+# Referred from transformers.models.bert.modeling_bert.BertEncoder
+class IBertEncoder(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.config = config
+        self.quant_mode = config.quant_mode
+        self.layer = nn.ModuleList([IBertLayer(config) for _ in range(config.num_hidden_layers)])
+
+    def forward(
+        self,
+        hidden_states,
+        hidden_states_scaling_factor,
+        attention_mask=None,
+        head_mask=None,
+        encoder_hidden_states=None,
+        encoder_attention_mask=None,
+        past_key_values=None,
+        use_cache=None,
+        output_attentions=False,
+        output_hidden_states=False,
+        return_dict=True,
+    ):
+        all_hidden_states = () if output_hidden_states else None
+        all_self_attentions = () if output_attentions else None
+        all_cross_attentions = () if output_attentions and self.config.add_cross_attention else None
```

Comment:
It doesn't seem like EncoderDecoder is supported yet -> so let's just remove this?